docs/src/main/asciidoc/kafka-reactive-getting-started.adoc

```diff
@@ -0,0 +1,365 @@
+The result is sent to an in-memory stream consumed by a JAX-RS resource.
+The data is sent to a browser using server-sent events.
+
+image::kafka-guide-architecture.png[alt=Architecture, align=center]
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `kafka-quickstart` {quickstarts-tree-url}/kafka-quickstart[directory].
+
+== Creating the Maven Project
+
+First, we need a new project.
+Create a new project with the following command:
+
+[source,bash,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=kafka-quickstart \
+    -DclassName="org.acme.kafka.PriceResource" \
+    -Dpath="/prices" \
+    -Dextensions="resteasy-reactive,resteasy-reactive-jackson,smallrye-reactive-messaging-kafka"
+cd kafka-quickstart
+----
+
+This command generates a Maven project, importing the Reactive Messaging and Kafka connector extensions.
+
+If you already have your Quarkus project configured, you can add the `smallrye-reactive-messaging-kafka`
+and `resteasy-jackson` extensions to your project by running the following command in your project base directory:
+
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="smallrye-reactive-messaging-kafka,resteasy-jackson"
+----
+
+This will add the following to your `pom.xml`:
+
+[source,xml]
+----
+
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
+</dependency>
+----
+
+[TIP]
+.Dev Services
+====
+No need to start a Kafka broker when using the dev mode or for tests.
+Quarkus starts a broker for you automatically.
+See xref:kafka-dev-services.adoc[Dev Services for Kafka] for details.
+====
+
+== The price object
+
+Create the `src/main/java/org/acme/kafka/Price.java` file, with the following content:
+
+[source,java]
+----
+package org.acme.kafka;
+
+public class Price {
+
+    public double value;
+    public String currency;
+
+    public Price() { }
+
+    public Price(double value, String currency) {
+        this.value = value;
+        this.currency = currency;
+    }
+}
+----
+
+JSON representation of `Price` objects will be used in messages sent to the Kafka topic
+and also server-sent events sent to browser clients.
+
+Quarkus has built-in capabilities to deal with JSON Kafka messages.
+In a following section we will create serializer/deserializer classes for Jackson.
+
+== The price generator
+
+Create the `src/main/java/org/acme/kafka/PriceGenerator.java` file, with the following content:
```

Comment:
We need to decide which approach we take. From the use case we had, we should use an emitter-based approach, with a simple endpoint.